Update changelog for v0.12 (#3665)
openenclave/openenclave: CHANGELOG.md

[Unreleased][Unreleased_log]
--------------

[v0.12.0][v0.12.0_log]
--------------

### Added
- Initial implementation of the [Malloc Info API](docs/DesignDocs/Mallinfo.md) for dlmalloc (default allocator), and snmalloc.
- Added missing attribute validations to oeedger8r C++ implementation.

> - Added new API *oe_log_message* see [design doc](docs/DesignDocs/oe_log_message()_callback_proposal.md) and [sample](samples/log_callback/README.md).

Reviewer: ```suggestion
- Added new API *oe_log_message*.  See [design doc](docs/DesignDocs/oe_log_message()_callback_proposal.md) and [sample](samples/log_callback/README.md).
```

### Changed

> - Fixed #3543, updated openenclaverc file and documents on Windows to avoid overwriting CMAKE_PREFIX_PATH.

Reviewer: ```suggestion
- Fixed https://github.com/openenclave/openenclave/issues/3543, updated openenclaverc file and documents on Windows to avoid overwriting CMAKE_PREFIX_PATH.
```

Once cloned locally, I believe the markdown won't automatically recognize issue numbers so any number should have a link, like lines 31, 35, etc do.

### Security
- Security fixes in oeedger8r
     - Fix TOCTOU vulnerability in NULL terminator checks for ocall in/out string parameters.
     - Count/size properties in deep-copied in/out structs are treated as read-only to prevent the host
	   from changing corrupting enclave memory by changing these properties.

> - Fixed bug where socket syscalls can leak enclave memory contents (CVE-2020-15224).

Author: ### Fixed [Socket syscalls can leak enclave memory contents](https://github.com/openenclave/openenclave/security/advisories/GHSA-525h-wxcc-f66m) (CVE-2020-15224).

     - Fix TOCTOU vulnerability in NULL terminator checks for ocall in/out string parameters.

>      - Count/size properties in deep-copied in/out structs are treated as readonly to prevent the host

Reviewer 2: Not part of these changes, but could be cleaned up.
```suggestion
     - Count/size properties in deep-copied in/out structs are treated as read-only to prevent the host
```

	   from changing corrupting enclave memory by changing these properties.

> - Fixed socket syscalls can leak enclave memory contents (CVE-2020-15224).

Reviewer: ```suggestion
- Fixed bug where socket syscalls can leak enclave memory contents (CVE-2020-15224).
```

> - Fixed socket syscalls can leak enclave memory contents (CVE-2020-15224).

Reviewer 2: I think if the intent is to use the advisory name, then it's clearer if we actually link the advisory here. 
```suggestion
- Fixed [Socket syscalls can leak enclave memory contents](https://github.com/openenclave/openenclave/security/advisories/GHSA-525h-wxcc-f66m) (CVE-2020-15224).
```

Author: Fixed.
